lib/modules/manager/gradle-wrapper/artifacts.ts

```diff
@@ -194,6 +194,48 @@ export async function updateArtifacts({
       );
     }
 
+    let writeLocksCmd = await prepareGradleCommand(gradlewFile);
+    if (!writeLocksCmd) {
+      logger.info('No gradlew found - skipping Artifacts update');
+      return null;
+    }
+    writeLocksCmd += ' :wrapper --write-locks';
+
+    logger.debug(`Downloading the new gradle wrapper and updating the locks: "${writeLocksCmd}"`);
+    try {
+      await exec(writeLocksCmd, execOptions);
+    } catch (err) {
+      // istanbul ignore if
+      if (err.message === TEMPORARY_ERROR) {
+        throw err;
+      }
+      logger.warn(
+        { err },
+        'Error executing gradle wrapper download and lock update command. It can be not a critical one though.',
+      );
+    }
+
+    let updateWrapperCmd = await prepareGradleCommand(gradlewFile);
+    if (!updateWrapperCmd) {
+      logger.info('No gradlew found - skipping Artifacts update');
+      return null;
+    }
+    updateWrapperCmd += ' :wrapper';
```

Comment:
Do I understand correctly that this function will now:
- Run gradlew :wrapper
- Run gradlew :wrapper --write-locks
- Run gradlew :wrapper again?

Even if this is correct, we need to document the reasoning behind this logic for future developers/maintainers

Reply:
Pending the resolution of the discussion in https://github.com/renovatebot/renovate/discussions/34273 the plan would be to run gradlew :wrapper twice. I can add in documentation explaining why that is the case.